### PR TITLE
DR-3027 create collection card component

### PR DESCRIPTION
**Pull request by avertrees**
## Ticket:

- JIRA ticket [DR-3027](https://newyorkpubliclibrary.atlassian.net/browse/DR-3027)

## This PR does the following:

- creates a CollectionCard component using the card in the swimLanes component
- creates a CollectionCard type interface and CollectionCardModel class (this may be a little redundant, thinking of refactoring later)
- creates LaneData type interface 

## How has this been tested?

locally by making sure cards load as expected

## Accessibility concerns or updates

- zilch

### Checklist:

- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.
- [x] I have updated the CHANGELOG.md.

[DR-3027]: https://newyorkpubliclibrary.atlassian.net/browse/DR-3027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by 7emansell — COMMENTED** [review]
Overall looks great, minor comments! Also I think there will be some style changes needed (particularly the `LargerThanLargeTablet` prop) once the card column breakpoints (#155) is merged in so maybe wait on that/ merge it in? It's still in VQA so I'm not sure how close that is to what we'll actually have

File: app/types/CollectionCard.ts
Comment (avertrees) [reply app/types/CollectionCard.ts:1]: True.. think it would be more clear but I was just following the pattern already in place with the following types: FeaturedItemData, ExploreFurtherData, and FeaturedContentData . If we add ...Type to the new types, I think we should also add them the existing ones - do you agree?

Comment (avertrees) [reply app/types/CollectionCard.ts:1]: also wondering if we should have `props` directory in the types folder

Comment (EdwinGuzman) [reply app/types/CollectionCard.ts:1]: Try the `props` directory? It's a reversible small update so you can always try it to see if it's helpful.

Personally, I like to keep the prop type/interface in the same file as the component for readability. It's the convention in the DS and maybe other projects.

File: app/components/cards/collectionCard.tsx
Comment (EdwinGuzman) [inline app/components/cards/collectionCard.tsx:10]: Ah, start creating separate CSS module files for each component. So there should be a new one for `CollectionCard` to better scope styles to each component.

File: app/types/CollectionCard.ts
Comment (7emansell) [reply app/types/CollectionCard.ts:1]: I agree we should add ...Type to everything. `props` directory is up to you if you think it'd be good @avertrees

File: app/components/cards/collectionCard.tsx
Comment (EdwinGuzman) [inline app/components/cards/collectionCard.tsx:79]: Remove the comments.

File: app/components/cards/collectionCard.tsx
Comment (EdwinGuzman) [inline app/components/cards/collectionCard.tsx:63]: Use the breakpoint values found in `app/utils/breakpoints.ts`. In this case I think this should be `lgMobile`.

Comment (EdwinGuzman) [reply app/components/cards/collectionCard.tsx:63]: The bigger problem is that `CollectionCard` is composed to replaced the `Card` component in `swimLanes.tsx` and so far it's coming along well. But when you get to replacing `Card` in `exploreFurther.tsx` with `CollectionCard`, it won't be a direct swap because the heading and content are _slightly_ different.

We can chat about some approaches tomorrow.

File: app/components/cards/collectionCard.tsx
Comment (avertrees) [reply app/components/cards/collectionCard.tsx:79]: oop

File: app/components/cards/collectionCard.tsx
Comment (avertrees) [reply app/components/cards/collectionCard.tsx:63]: I don't think CollectionCard should be used to replace the Card component in exploreFurther - I wasn't planning on swapping them out either but if we want to make a new ticket for that we can.

Comment (avertrees) [reply app/components/cards/collectionCard.tsx:63]: I think CollectionCard should represent a Collection - the Explore Further section does not display Collections

**Review by EdwinGuzman — COMMENTED** [review]
Latest round of updates look good but create a new CSS modules file for `CollectionCard` component.

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**
